File: typescript-client-sdk/src/sdk/models/operations/getintegrationsbanksettings.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import { Type } from "class-transformer";


export class GetIntegrationsBankSettingsSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetIntegrationsBankSettingsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  security: GetIntegrationsBankSettingsSecurity;
}

export class GetIntegrationsBankSettingsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatPublicApiModelsClientsBankingSettingsModelsBankSettingsDataset?: shared.CodatPublicApiModelsClientsBankingSettingsModelsBankSettingsDataset;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: python-client-sdk/src/sdk/models/operations/get_integrations_banksettings.py
import dataclasses
from typing import Optional
from ..shared import security as shared_security
from ..shared import security as shared_security
from ..shared import codat_public_api_models_clients_bankingsettingsmodels_banksettingsdataset as shared_codat_public_api_models_clients_bankingsettingsmodels_banksettingsdataset


@dataclasses.dataclass
class GetIntegrationsBankSettingsSecurity:
    api_key_auth: Optional[shared_security.SchemeAPIKeyAuth] = dataclasses.field(default=None, metadata={'security': { 'scheme': True, 'type': 'apiKey', 'sub_type': 'header' }})
    codat_login: Optional[shared_security.SchemeCodatLogin] = dataclasses.field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclasses.dataclass
class GetIntegrationsBankSettingsRequest:
    security: GetIntegrationsBankSettingsSecurity = dataclasses.field()
    

@dataclasses.dataclass
class GetIntegrationsBankSettingsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    codat_public_api_models_clients_banking_settings_models_bank_settings_dataset: Optional[shared_codat_public_api_models_clients_bankingsettingsmodels_banksettingsdataset.CodatPublicAPIModelsClientsBankingSettingsModelsBankSettingsDataset] = dataclasses.field(default=None)
    
